Metalama.Framework/Diagnostics/SuppressionDiagnostic.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+using System.Collections.Immutable;
+
+namespace Metalama.Framework.Diagnostics;
+
```

Comment:
Missing XML doc.

Reply:
Change the `struct` to an `interface` named `ISuppressibleDiagnostic` with an internal implementation in the Engine project and implement lazy (Memo) implementation of the different properties.